src/commands/lint/install.rs

```diff
@@ -0,0 +1,210 @@
+use anyhow::Context;
+use reqwest::blocking::Client;
+use serde::Deserialize;
+#[cfg(feature = "rustup")]
+use tracing::debug;
+
+#[cfg(feature = "rustup")]
+use crate::{
+    commands::lint::InstallArgs,
+    external_cli::{CommandExt, cargo::install::AutoInstall},
+};
+#[derive(Deserialize, Debug)]
+#[allow(dead_code)]
+struct RustToolchain {
+    toolchain: Toolchain,
+}
+
+#[derive(Deserialize, Debug)]
+#[allow(dead_code)]
+struct Toolchain {
+    channel: String,
+    components: Vec<String>,
+}
+
+#[cfg(feature = "rustup")]
+pub(crate) fn install_linter(arg: &InstallArgs) -> anyhow::Result<()> {
+    use std::env;
+
+    const GIT_URL: &str = "https://github.com/TheBevyFlock/bevy_cli.git";
+
+    // Get a list of all available `bevy_lint` versions, there should always be at least one (main).
+    let available_versions = list_available_releases()?;
+
+    // A specific version was passed in the `InstallArgs`
+    let (rust_toolchain, version) = if let Some(version) = &arg.version {
+        // Check if the desired version exists, if not return with an error message
+        if !available_versions.contains(version) {
+            anyhow::bail!(
+                "version `{}` does not exist. Available versions: {:?}",
+                version,
+                available_versions
+            );
+        }
+        // Return the required toolchain version and the name of the linter tag or `main` that
+        // corresponds to the desired version.
+        (lookup_toolchain_version(version)?, version)
+    }
+    // No version was passed in the `InstallArgs` open a dialog with all available versions
+    // (including the main branch) to choose from.
+    else {
+        let Some(selection) = dialoguer::FuzzySelect::new()
+            .with_prompt("Available `bevy_lint` versions")
+            .items(&available_versions)
+            .interact_opt()?
+        else {
+            debug!("no version selected");
+            return Ok(());
+        };
+
+        let version = &available_versions[selection];
+        debug!("selected {}", version);
+
+        let required_toolchain = lookup_toolchain_version(version)?;
+
+        // If no specific version was passed, ask for confirmation.
+        if !dialoguer::Confirm::new()
+            .with_prompt(format!(
+                "Do you want to install `bevy_lint-{version}` and the required toolchain: `{}` ?",
+                required_toolchain.toolchain.channel
+            ))
+            .interact()
+            .context(
+                "failed to show interactive prompt, try passing a specific version as an argument",
+            )?
+        {
+            anyhow::bail!(
+                "User does not want to install `bevy_lint-{version}` and the required toolchain: `{}`",
+                required_toolchain.toolchain.channel
+            );
+        }
```

Comment:
I think this dialogue should be shown even if the user installs a specific version (e.g. `bevy lint install v0.4.0`), but it should also respect `--yes` if passed. This is mainly so the user can verify the Rustup toolchain is desired